### PR TITLE
wrap all callbacks in instrumentations with wrapFunction

### PR DESCRIPTION
**Pull request by bengl**
These were the remaining areas with non-trivial code that might not be caught by try/catches when our code throws an exception. Adding `wrapFunction` adds that try/catch appropriately, but only when SSI is enabled. Otherwise these are all passthroughs. In the future, we may enable those try/catches at all times.

I considered the following to be out of scope:

* Many (but not all) completely trivial wrappers.
* Exceptions that might originate in `async_hooks`.

I may have missed some. It's an extremely difficult task to verify that all opportunities for throwing an exception are covered.

### CODE REVIEW

**Review by Qard — DISMISSED** [review]

File: packages/datadog-instrumentations/src/winston.js [hunk 1]
```diff
@@ -43,12 +43,12 @@ function wrapMethod (method, logCh) {
         if (patched.has(transport) || typeof transport.log !== 'function') continue
 
         const log = transport.log
-        transport.log = function wrappedLog (level, msg, meta, callback) {
+        transport.log = shimmer.wrapFunction(log, log => function wrappedLog (level, msg, meta, callback) {
```
Comment (simon-id) [inline packages/datadog-instrumentations/src/winston.js:46]: shouldn't this be 
```javascript
shimmer.wrap(transport, 'log', ...)
```
instead ?

File: packages/datadog-instrumentations/src/mongoose.js [hunk 2]
```diff
@@ -128,22 +128,21 @@ addHook({
                     const resolve = arguments[0]
                     const reject = arguments[1]
 
-                    // not using shimmer here because resolve/reject could be empty
-                    arguments[0] = function wrappedResolve () {
```
Comment (simon-id) [inline packages/datadog-instrumentations/src/mongoose.js:132]: it says it doesn't use shimmer for a specific reason here, is that handled correctly by shimmer now ?

File: packages/datadog-instrumentations/src/winston.js (hunk 1 above)
Comment (bengl) [reply packages/datadog-instrumentations/src/winston.js:46]: It probably always should have been `shimmer.wrap()` yeah. I can change this one.

Comment (bengl) [reply packages/datadog-instrumentations/src/winston.js]: Done.

File: packages/datadog-instrumentations/src/grpc/client.js [hunk 3]
```diff
@@ -88,10 +88,10 @@ function wrapMethod (method, path, type, hasPeer) {
...
   }
 
-  const wrapped = function () {
+  const wrapped = shimmer.wrapFunction(method, method => function () {
     const args = ensureMetadata(this, arguments, 1)
     return callMethod(this, method, args, path, args[1], type, hasPeer)
-  }
+  })
 
   Object.assign(wrapped, method)
```
[the remark this answers is not in the document]
Comment (bengl) [reply packages/datadog-instrumentations/src/grpc/client.js]: Fixed.

**Review by simon-id — APPROVED** [review]

**Comment by bengl** [comment]
The two failing are all-green and next.js, so I'll just go ahead and merge.

**Pull request merged**